filetests/isa/x86/legalize-splat.clif

```diff
@@ -0,0 +1,72 @@
+test compile
+set probestack_enabled=false
+target x86_64 icelake
+
+; use baldrdash calling convention here for simplicity (avoids prologue, epilogue)
+function %test_splat_i32() -> i32x4 baldrdash {
+ebb0:
+    v0 = iconst.i32 42
+    v1 = splat.i32x4 v0
+    return v1
+}
+
+; sameln: function %test_splat_i32() -> i32x4 [%xmm0] baldrdash {
+; nextln:   ss0 = incoming_arg 0, offset 0
+; nextln: 
+; nextln:   ebb0:
+; nextln:     v0 = iconst.i32 42
+; nextln:     v2 = x86_scalar_to_vector.i32x4 v0
+; nextln:     v1 = x86_pshufd v2, 0
+; nextln:     return v1
+; nextln: }
+
+
+
+function %test_splat_i64() -> i64x2 baldrdash {
+ebb0:
+    v0 = iconst.i64 42
+    v1 = splat.i64x2 v0
+    return v1
+}
+
+; check:   ebb0:
+; nextln:     v0 = iconst.i64 42
+; nextln:     v2 = x86_scalar_to_vector.i64x2 v0
+; nextln:     v1 = insertlane v2, 1, v0
+; nextln:     return v1
+
+
+
+function %test_splat_b16() -> b16x8 baldrdash {
+ebb0:
+    v0 = bconst.b16 true
+    v1 = splat.b16x8 v0
+    return v1
+}
+
+; check:   ebb0:
+; nextln:     v0 = bconst.b16 true
+; nextln:     v2 = x86_scalar_to_vector.b16x8 v0
+; nextln:     v3 = insertlane v2, 1, v0
+; nextln:     v4 = bitcast.i32x4 v3
+; nextln:     v5 = x86_pshufd v4, 0
+; nextln:     v1 = bitcast.b16x8 v5
+; nextln:     return v1
```

Comment:
All of the other blocks in this file seem to work but this one fails with:
```
FAIL filetests/isa/x86/legalize-splat.clif: compile(%test_splat_b16):
function %test_splat_b16() -> b16x8 [%xmm0] baldrdash {
                                ebb0:
[-]                                 v0 = bconst.b16 true
[RexMp2frurm#56e]                   v2 = x86_scalar_to_vector.b16x8 v0
[Mp2r_ib_unsigned_r#5c4]            v3 = insertlane v2, 1, v0
[null_fpr#00]                       v4 = bitcast.i32x4 v3
;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
; error: inst5: arg 0 (v3) with type b16x8 failed to satisfy type set ValueTypeSet { lanes: BitSet(511), ints: BitSet(120), floats: BitSet(96), bools: BitSet(0) }

[Mp2r_ib_unsigned#570]              v5 = x86_pshufd v4, 0
[null_fpr#00]                       v1 = bitcast.b16x8 v5
;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
; error: inst1: has an invalid controlling type b16x8

[Op1ret#c3]                         return v1
}

; 2 verifier errors detected (see above). Compilation aborted.
```

I'm pretty sure I've created the right bitcasts at https://github.com/CraneStation/cranelift/pull/833/files#diff-d8dea7e9b6533024c2c26fbb8d88867bR1654 and following; the legalization for this is at https://github.com/CraneStation/cranelift/pull/833/files#diff-4643787d63e13b9a6b517d0bebb29decR346.

Reply:
Thanks for the help, @sunfishcode! Should be fixed now with `raw_bitcast`